src/components/Onboard/SerialPortOption.tsx

```diff
@@ -1,37 +1,70 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
 import {
   RadioIcon,
   CheckCircleIcon,
   XCircleIcon,
 } from "@heroicons/react/24/outline";
+import { requestConnectToDevice } from "@app/features/device/deviceActions";
+import { selectDeviceConnected } from "@app/features/device/deviceSelectors";
+import { selectRequestStateByName } from "@app/features/requests/requestSelectors";
 
 export interface ISerialPortOptions {
   name: string;
   state: string;
-  connection_error: string;
+  isActive: boolean;
+  onSelection: () => void;
 }
 
 const SerialPortOption = ({
   name,
   state,
-  connection_error,
+  isActive,
+  onSelection,
 }: ISerialPortOptions) => {
-  switch (state) {
+  const dispatch = useDispatch();
+  const isDeviceConnected = useSelector(selectDeviceConnected());
+  const statusOfConnect = useSelector(selectRequestStateByName(name));
```

Comment:
What is this `name` field being used for? I was imagining you would use `useSelector(selectRequestStateByName(requestConnectToDevice.type))`

Reply:
This was passed in as a prop to identify which component is currently being handled. I've updated it from name to your implementation.